recipe/builder.py

```diff
@@ -14,13 +14,25 @@ def install():
     """ populate the environment variable added by the patch and actually
         install with pip
     """
+
+    cuda_version = os.environ.get('cuda_compiler_version', None)
+
+    try:
+        float(cuda_version)
+        is_cuda = True
+    except (TypeError, ValueError):
+        is_cuda = False
+
     if WIN:
         cmake_args = (
             "-DDLIB_FORCE_MSVC_STATIC_RUNTIME=OFF\n"
             "-DDLIB_LINK_WITH_SQLITE3=OFF\n"
             "-DBUILD_SHARED_LIBS=ON\n"
             "-DDLIB_IN_PROJECT_BUILD=ON\n"
-        ).format(os.environ)
+            "-DCUDNN_INCLUDE_DIR:PATH={LIBRARY_INC}\n"
+            "-DCUDNN_ROOT:PATH={LIBRARY_PREFIX}\n"
```

Comment:
Maybe we can do something like this for finding the CUDA Toolkit

```suggestion
            "-CUDAToolkit_ROOT:PATH="
            "{LIBRARY_PREFIX}" if cuda_version >= 12.0 else "{CUDA_HOME}"
            "\n"
            "-DCUDNN_ROOT:PATH={LIBRARY_PREFIX}\n"
```